File: linkedlist/golang/linked_list.go
package main

import "errors"

type Node struct {
	value interface{}
	next *Node
}

func newNode(v interface{}) *Node{
	return &Node{value: v}
}

type LinkedList struct {
	head *Node
}

func newLinkedList() LinkedList {
	return LinkedList{}
}

func (l LinkedList) pushFront(v interface{}){

	node := newNode(v)

	if(l.head != nil){
		node.next = l.head
	}

	l.head = node
}

func (l LinkedList) front() (interface{}, error){

	if(l.head == nil){
		return nil, errors.New("List is empty")
	}

	return l.head.value, nil
}

File: linkedlist/cpp/src/linked_list.cpp
#include "linked_list.h"
#include <stdexcept>

namespace jg {

    template<class T>
    bool LinkedList<T>::Empty() const {
        return head_ == nullptr;
    }

    template <class T>
    void LinkedList<T>::PushFront(T value) {
        std::unique_ptr<ListElement<T>> node(new ListElement<T>(value));

        node->next = std::move(head_);
        head_ = std::move(node);
    }

    template <class T>
    void LinkedList<T>::PushBack(T value) {
        std::unique_ptr<ListElement<T>> newNode(new ListElement<T>(value));

        if (Empty()){
            head_ = std::move(newNode);
            return;
        }

        auto node = head_.get();

        while(node->next != nullptr){
            node = node->next.get();
        }

        node->next = std::move(newNode);
    }

    template <class T>
    T LinkedList<T>::Front() {
        if (Empty()){
            throw std::logic_error("List is empty");
        }

        return head_->value;
    }

    template <class T>
    T LinkedList<T>::Back() {
        if (Empty()){
            throw std::logic_error("List is empty");
        }

        auto node = head_.get();

        while(node->next != nullptr){
            node = node->next.get();
        }

        return node->value;
    }
}